websites/B/Baseball-Reference/presence.ts

```diff
@@ -0,0 +1,92 @@
+	if (hostname === "baseball-reference.com") {
+		if (pathname === "/") presenceData.details = "Viewing Home Page";
+	} else if (pathname.includes("/players")) {
+		presenceData.details = "Viewing a Player";
+		presenceData.state = `${
+			document.querySelector("div#meta div h1 span").textContent
+		} | ${document
+			.querySelector("div#meta div p strong")
+			.nextSibling.textContent.replace(/\s/g, " ")
+			.trim()}`;
+		if (!presence.getActivity().largeImageKey)
+			presenceData.largeImageKey = image;
+
+		presenceData.buttons = [{ label: "View Player", url: href }];
+
+		if (
+			document.querySelector("#meta > div:nth-child(2) > p:nth-child(5) > a")
+		) {
+			team = document.querySelector(
+				"#meta > div:nth-child(2) > p:nth-child(5) > a"
+			).textContent;
+			presenceData.state = `${presenceData.state} For The ${team}`;
+		}
+	} else if (pathname.includes("/managers")) {
+		presenceData.details = "Viewing a Manager";
+		presenceData.state = document.querySelector(
+			"div#meta div h1 span"
+		).textContent;
+		if (!presence.getActivity().largeImageKey)
+			presenceData.largeImageKey = image;
+
+		presenceData.buttons = [{ label: "View Manager", url: href }];
+	} else if (pathname.includes("/teams")) {
+		presenceData.details = "Viewing a Team";
+		presenceData.state = `${
+			document.querySelector("div#meta div h1 span").textContent
+		} ${
+			document.querySelector("div#meta div h1 span").nextElementSibling
+				.textContent
+		}`;
+		if (!presence.getActivity().largeImageKey)
+			presenceData.largeImageKey = image;
+
+		presenceData.buttons = [{ label: "View Team", url: href }];
+	} else if (pathname.includes("/leagues")) {
+		presenceData.details = "Viewing Stats";
+		presenceData.state = `${
+			document.querySelector("div#meta div h1 span").textContent
+		} ${
+			document.querySelector("div#meta div h1 span").nextElementSibling
+				.textContent
+		} ${document.querySelector("div#meta div h1 span.header_end").textContent}`;
+		if (!presence.getActivity().largeImageKey)
+			presenceData.largeImageKey = image;
+
+		presenceData.buttons = [{ label: "View Page", url: href }];
+	} else if (pathname.includes("/boxes")) {
+		presenceData.details = "Viewing Box Score";
+		presenceData.state = document
+			.querySelector("div h1")
+			.textContent.replace(" vs ", "@");
+		if (!presence.getActivity().largeImageKey)
+			image = document.images.item(2).src;
+
+		presenceData.largeImageKey = image;
+		presenceData.buttons = [{ label: "View Box Score", url: href }];
+	} else if (pathname.includes("/friv")) {
+		presenceData.details = "Viewing Misc";
+		presenceData.state = document.querySelector("div h1").textContent;
+		presenceData.buttons = [{ label: "View Page", url: href }];
+	} else if (pathname.includes("/oracle")) {
+		presenceData.details = "Consulting the Oracle of Baseball";
+		presenceData.state = document.querySelector("div h1").textContent;
+		presenceData.buttons = [{ label: "View Page", url: href }];
```

Comment:
It seems like there may be a chance no largeImage is shown, you might want to change it so it always has a default